packages/js/src/generators/library/library.ts

```diff
@@ -177,6 +179,39 @@ function updateTsConfig(tree: Tree, options: NormalizedSchema) {
   });
 }
 
+/**
+ * Currently `@nrwl/js:library` TypeScript files can be compiled by most NX applications scaffolded via the Plugin system. However, `@nrwl/react:app` is an exception that due to its babel configuration, won't transpile external TypeScript files from packages/libs that do not contain a .babelrc.
+ * To prevent breaking the experience for the user (they see the application failing, and they need to manually add the babelrc themselves), we want to detect whether they have the `@nrwl/web` plugin installed, and generate it automatically for them (even when they do not explicity request it).
+ *
+ * You can find more details on why this is necessary here:
+ * https://github.com/nrwl/nx/pull/10055
+ */
+function shouldAddBabelRc(tree: Tree, options: NormalizedSchema) {
```

Comment:
question: what if I have `nrwl/web` installed and I want to generate a `nrwl/js:lib` BUT I do not want to have `.babelrc` generated by explicitly use `--includeBabelRc=false`?

Thinking of a monorepo with both `web` and an `api`

Reply:
@nartc good point, I just pushed a refactor and added unit tests to cover that scenario, so that now we correctly handle

1. user explicitly setting the flag to `true`
2. user explicitly setting the flag to `false`
3. user not setting the flag 

I feel we landed on a good implementation now, let me know your thoughts

Reply:
You can just remove the `default` in the schema and it will come through as `undefined` (as long as the flag isn't in the `required` array in the schema)